### DISCUSSION TITLE
Lack of Right-to-Left (RTL) Language Support in ConnectKit

### DISCUSSION THREAD

**Discussion by nekofar** [family/connectkit#305] [Ideas]
**Describe the bug**

When using ConnectKit, I noticed that the application does not correctly support right-to-left (RTL) languages. This results in incorrect text rendering and layout issues when the application is used in an RTL language environment.

**To reproduce**

The version of ConnectKit: 1.5.1

Steps to reproduce:
1. Launch the ConnectKit Nex.js app.
2. You have two options to change the `dir` attribute of the `html` element to `rtl`. You can either use Developer Tools or add a `_document.tsx` file to the project and modify the code directly from there.
4. Navigate to the `Connect Wallet` button on the index and click on it.
5. Please take note of a blank modal that appears on your computer's desktop.

**Expected behavior**

I expected the ConnectKit application to properly support right-to-left languages properly, ensuring correct text rendering and layout alignment. All text elements, buttons, and UI components should be mirrored and aligned according to the RTL language standards.

**Screenshots**

| LTR | RTL |
|--------|--------|
| ![Capture-2023-08-16-144932](https://github.com/family/connectkit/assets/147401/de309cd2-d575-4a85-89b9-1547fa2f19c0) | ![Capture-2023-08-16-145044](https://github.com/family/connectkit/assets/147401/8523cad3-49f4-45a5-a374-1ba06a5b7523) | 

**Environment details**

- Operating System: macOS
- Browser: Google Chrome
- Browser Version: 115.0.5790.170
- Device: Desktop

**Additional context**

It's important to note that supporting right-to-left languages is crucial for ensuring a seamless experience for users who rely on these languages. Proper RTL language support includes text alignment and considerations for overall layout, padding, and visual elements.

This bug report highlights the current lack of RTL language support in ConnectKit and emphasizes the importance of addressing this issue to provide a more inclusive user experience. Please don't hesitate to contact me if you need further information or help reproduce the problem.

**Comment by nekofar** [comment]
https://github.com/family/connectkit/blob/47984040867a15ff8cbfdcdea534ad662c2d405e/packages/connectkit/src/components/Common/Modal/styles.ts#L311-L312

While this hack could address the problem of the empty modal, a more comprehensive solution that fully supports RTL is needed.

```
left: ${document.documentElement.dir === 'rtl' ? 'auto' : '50%'};
right: ${document.documentElement.dir === 'rtl' ? '50%' : 'auto'};
margin-left: ${document.documentElement.dir === 'rtl' ? 'auto' : 'calc(var(--width) / -2)'};
margin-right: ${document.documentElement.dir === 'rtl' ? 'calc(var(--width) / -2)' : 'auto'};
```

![Capture-2023-08-16-172805](https://github.com/family/connectkit/assets/147401/c964a625-c254-454e-8780-ccd0d9cf0382)

**Comment by nekofar** [comment]
I haven't personally used `styled-components`, so I'm not very familiar with it. However, it seems that there is an [official plugin](https://github.com/styled-components/stylis-plugin-rtl) available to support RTL.

**Comment by Javid-Izadfar** [comment]
fixed the issue on my project by simply changing the direction at some global .css file.

```
#__CONNECTKIT__ {
  direction: ltr;
}
```
